### PR TITLE
FIX-#7051: Update exception message for `astype` function

### LINKED ISSUE #7051: Update Exception message for `astype` function in the case of duplicated values

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

Part of https://github.com/modin-project/modin/pull/6954

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7051 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -900,10 +900,18 @@ def execute_callable(fn, inplace=False, md_kwargs={}, pd_kwargs={}):
                 ), "Got Modin Exception type {}, but pandas Exception type {} was expected".format(
                     type(md_e), type(pd_e)
                 )
-                if raising_exceptions:
+                if raising_exceptions and isinstance(raising_exceptions, (list, tuple)):
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:903]: This code branch is temporary and will be removed in https://github.com/modin-project/modin/pull/6954.

File: modin/pandas/test/utils.py [hunk 2]
```diff
@@ -900,10 +900,18 @@ def execute_callable(fn, inplace=False, md_kwargs={}, pd_kwargs={}):
...
                     assert not isinstance(
                         md_e, tuple(raising_exceptions)
                     ), f"not acceptable exception type: {md_e}"
+                elif raising_exceptions and type(raising_exceptions) is type:
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:907]: `type(raising_exceptions) is type:` - сhecking to ensure that it is not the class that is being transferred, but its instance

**Comment by anmyachev** [comment]
@YarShev please take a look

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (YarShev) [reply modin/pandas/test/utils.py:903]: How will this complicate further development until #6954 is merged?

File: modin/pandas/test/utils.py (hunk 2 above)
Comment (YarShev) [reply modin/pandas/test/utils.py:907]: I don't quite get this. Could you provide an example of passing such an instance?

Comment (anmyachev) [reply modin/pandas/test/utils.py:907]:
```python
>>> type(TypeError) is type
True
>>> type(TypeError("something")) is type 
False

```

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/utils.py:903]: In general, it shouldn’t complicate things in any way, since this parameter is still optional.

**Review by YarShev — APPROVED** [review]

**Pull request merged**